cli/cmd/generate_gcp.go

```diff
@@ -0,0 +1,780 @@
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.BucketLocation,
+		"bucket_location",
+		"",
+		"specify gcp bucket location")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.BucketName,
+		"bucket_name",
+		"",
+		"specify new bucket name")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.ExistingLogBucketName,
+		"existing_bucket_name",
+		"",
+		"specify existing bucket name")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.ExistingLogSinkName,
+		"existing_sink_name",
+		"",
+		"specify existing sink name")
+	generateGcpTfCommand.PersistentFlags().BoolVar(
+		&GenerateGcpCommandState.EnableForceDestroyBucket,
+		"enable_force_destroy_bucket",
+		false,
+		"enable force bucket destroy")
+	generateGcpTfCommand.PersistentFlags().BoolVar(
+		&GenerateGcpCommandState.EnableUBLA,
+		"enable_ubla",
+		false,
+		"enable universal bucket level access(ubla)")
+	generateGcpTfCommand.PersistentFlags().IntVar(
+		&GenerateGcpCommandState.LogBucketLifecycleRuleAge,
+		"bucket_lifecycle_rule_age",
+		-1,
+		"specify the lifecycle rule age")
+	generateGcpTfCommand.PersistentFlags().IntVar(
+		&GenerateGcpCommandState.LogBucketRetentionDays,
+		"bucket_retention_days",
+		0,
+		"specify the bucket retention days")
+	generateGcpTfCommand.PersistentFlags().StringVar(
+		&GenerateGcpCommandState.AuditLogIntegrationName,
+		"audit_log_integration_name",
+		"",
+		"specify a custom audit log integration name")
+
+}
+
+func validateServiceAccountCredentialsFile(credFile string) error {
+	if file.FileExists(credFile) {
+		jsonFile, err := os.Open(credFile)
+		// if we os.Open returns an error then handle it
+		if err != nil {
+			return errors.Wrap(err, "Issue opening GCP credentials file")
+		}
+		defer jsonFile.Close()
+
+		byteValue, _ := ioutil.ReadAll(jsonFile)
+
+		var result map[string]interface{}
+		err = json.Unmarshal(byteValue, &result)
+		if err != nil {
+			return errors.Wrap(err, "Unable to parse credentials file.")
+		}
+		valid := validateSaCredFileContent(result)
+		if !valid {
+			return errors.New("Invalid GCP Service Account credentials file. " +
+				"The private_key and client_email fields MUST be present. private_key must be base64 encoded")
+		}
+
+	}
+	return nil
+}
+
+func validateSaCredFileContent(credFileContent map[string]interface{}) bool {
+	if credFileContent["private_key"] != nil && credFileContent["client_email"] != nil {
+		err := validateStringIsBase64(credFileContent["private_key"].(string))
+		if err == nil {
+			return true
+		}
+	}
+	return false
+}
+
+// create survey.Validator for string is base64
+func validateStringIsBase64(val interface{}) error {
+	switch value := val.(type) {
+	case string:
+		// if value isn't base64, return error
+		_, err := base64.StdEncoding.DecodeString(value)
+		if err != nil {
+			return errors.Wrap(err, "provided private key is not base64 encoded")
```

Comment:
Here an example of an error message starting with lowercases, lets update all errors to follow this pattern